src/vmm/src/lib.rs

```diff
@@ -157,8 +163,12 @@ impl TryFrom<VMMConfig> for VMM {
             kernel_cfg: config.kernel_config.clone(),
         };
 
-        vmm.add_serial_console()?;
         vmm.create_vcpus(&config.vcpu_config)?;
+        vmm.add_serial_console()?;
+
+        if config.block_config.is_some() {
+            vmm.add_block_device(&config.block_config.unwrap()).unwrap();
```

Comment:
We shouldn't unwrap here and propagate the error. This allows us to write negative tests as well as printing nice errors to the user.

Reply:
sure